Copy che-rest-apis code from che-workspace-operator

amisevsk opened the pull request:
### Description
This PR copies the che-rest-apis subproject from https://github.com/che-incubator/che-workspace-operator to this repo, using the commit https://github.com/che-incubator/che-workspace-operator/commit/ec1d080aa35086e7313cbaccdb3a3daa061a2215 as a base. It consists of three commits:

1. Copy in code without additional changes
2. Add README.md and dockerfile used for building without installing GraalVM (which can be tricky as the version is old and documentation is nowhere)
3. Clean up the codebase slightly.

amisevsk committed:
Copy che-rest-apis from che-workspace-operator repo

Copy the java codebase from
https://github.com/che-incubator/che-workspace-operator/

Current state is consistent with commit e767db9 in original repo

Signed-off-by: Angel Misevski <amisevsk@redhat.com>

amisevsk committed:
Add readme and a dockerfile for building without installing GraalVM

Signed-off-by: Angel Misevski <amisevsk@redhat.com>

amisevsk committed:
Clean up unused code and broken tests

- Add license headers to all files
- Clean up unused code and disable nonfunctional tests
- Add build.sh script to ease building final image

Signed-off-by: Angel Misevski <amisevsk@redhat.com>

amisevsk commented:
An image built from this PR is available at `amisevsk/che-rest-apis:latest`

sleshchenko commented on .dockerignore:
```diff
@@ -0,0 +1,4 @@
+# Note: this dockerignore is used by both the native binary and final container builds
+target
+!target/*-runner
+!target/libsunec.so
```
```suggestion
!target/libsunec.so

```

sleshchenko commented on src/main/java/org/eclipse/che/incubator/crd/cherestapis/ApiResource.java:
```diff
@@ -0,0 +1,60 @@
+        WorkspaceDto update) {
+      return addCorsHeader(apiService.getWorkspace(id));
+    }  
+}
```
```suggestion
}

```

sleshchenko commented on src/test/java/org/eclipse/che/incubator/crd/cherestapis/ApiResourceTest.java:
```diff
@@ -0,0 +1,33 @@
+          .then()
+             .statusCode(404);
+    }
+}
```
```suggestion
}

```

sleshchenko commented on src/test/java/org/eclipse/che/incubator/crd/cherestapis/ApiServiceTest.java:
```diff
@@ -0,0 +1,57 @@
+        WorkspaceDto workspace = service.convertToWorkspace(devfile, null);
+        Assertions.assertEquals(workspace.getConfig().getName(), "petclinic");
+    }
+}
```
```suggestion
}

```

amisevsk committed:
Avoid copying entire directory into docker daemon

The full target folder can be hundreds of megabytes in size; add a
dockerignore to only copy required pieces into the docker context

Signed-off-by: Angel Misevski <amisevsk@redhat.com>
